src/core/observer/index.js

```diff
@@ -172,7 +172,11 @@ export function defineReactive (
     set: function reactiveSetter (newVal) {
       const value = getter ? getter.call(obj) : val
       /* eslint-disable no-self-compare */
-      if (newVal === value || (newVal !== newVal && value !== value)) {
+      if (
+        (getter && !setter) ||
```

Comment:
Then when have the chance to invoke customSetter?

Reply:
@dsonet 

**The following content is my understanding, there may be incorrect place.**

The first thing we can guarantee is that if the value of the `property` property is `undefined`, both the value of `getter` and `setter` will be `undefined`:
```js
/* defineReactive */
const property = Object.getOwnPropertyDescriptor(obj, key)
// If `property === undefined`, both `setter` and `getter` are undefined
const getter = property && property.get
const setter = property && property.set
```
The `customSetter` method is mainly used to print prompt information when the property value is set, and mainly used in the following aspects:

* Define the injected value on the instance object:

```js
/* inject.js  initInjections */
defineReactive(vm, key, result[key], customSetter)
```
At this point, the return value of `Object.getOwnPropertyDescriptor(vm, key)` is `undefined`, so both `getter` and `setter` are `undefined`

* Define the `$attrs` and `$listeners` properties on the instance object:

```js
/* render.js  initRender */
    defineReactive(vm, '$attrs', parentData && parentData.attrs || emptyObject, () => {
      !isUpdatingChildComponent && warn(`$attrs is readonly.`, vm)
    }, true)
    defineReactive(vm, '$listeners', options._parentListeners || emptyObject, () => {
      !isUpdatingChildComponent && warn(`$listeners is readonly.`, vm)
    }, true)
```

We can still guarantee that the following statement's value is `undefined`:

```js
Object.getOwnPropertyDescriptor(vm, '$attrs') // undefined
Object.getOwnPropertyDescriptor(vm, '$listeners') // undefined
```

* Define the props data on the `vm._props` object:

```js
/* state  initProps */
const props = vm._props = {}
defineReactive(props, key, value, customSetter)
```

The value of `Object.getOwnPropertyDescriptor(props, key)` is `undefined`。

In summary, when the value of the `Object.getOwnPropertyDescriptor method` is `undefined`, both the `getter` and the `setter` are `undefined`, so `(getter && !setter)` will always be `false` and will not affect the execution of the `customSetter`.

Reply:
you are right

Reply:
@sodatea I have made some changes.